modules/tensor_mechanics/src/materials/CreepStressUpdateBase.C

```diff
@@ -0,0 +1,66 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CreepStressUpdateBase.h"
+
+template <>
+InputParameters
+validParams<CreepStressUpdateBase>()
+{
+  InputParameters params = validParams<RadialReturnStressUpdate>();
+
+  params.addCoupledVar("temperature", 0.0, "Coupled temperature");
+  params.addDeprecatedParam<std::string>(
+      "creep_prepend",
+      "",
+      "String that is prepended to the creep_strain Material Property",
+      "This has been replaced by the 'base_name' parameter");
+  params.set<std::string>("effective_inelastic_strain_name") = "effective_creep_strain";
+
+  return params;
+}
+
+CreepStressUpdateBase::CreepStressUpdateBase(const InputParameters & parameters)
+  : RadialReturnStressUpdate(parameters),
+    _creep_prepend(getParam<std::string>("creep_prepend")),
+    _has_temp(isCoupled("temperature")),
+    _temperature(_has_temp ? coupledValue("temperature") : _zero),
+    _creep_strain(declareProperty<RankTwoTensor>(_base_name + _creep_prepend + "creep_strain")),
+    _creep_strain_old(
+        getMaterialPropertyOld<RankTwoTensor>(_base_name + _creep_prepend + "creep_strain"))
+{
+}
+
+void
+CreepStressUpdateBase::initQpStatefulProperties()
+{
+  _creep_strain[_qp].zero();
+
+  RadialReturnStressUpdate::initQpStatefulProperties();
+}
+
+void
+CreepStressUpdateBase::propagateQpStatefulProperties()
+{
+  _creep_strain[_qp] = _creep_strain_old[_qp];
+
+  propagateQpStatefulPropertiesRadialReturn();
+}
+
+Real
+CreepStressUpdateBase::computeStressDerivative(const Real effective_trial_stress, const Real scalar)
+{
+  return -(computeDerivative(effective_trial_stress, scalar) + 1.0) / _three_shear_modulus;
+}
+
+void
+CreepStressUpdateBase::computeStressFinalize(const RankTwoTensor & plasticStrainIncrement)
+{
+  _creep_strain[_qp] += plasticStrainIncrement;
```

Comment:
@bwspenc This seems error prone to me. Shouldn't this just be `_creep_strain[_qp] += _creep_strain_old[_qp] + plasticStrainIncrement`?

Reply:
You are correct.  If you look at the BISON Zry creep models, they do it the way you say. Also, change the name `plasticStrainIncrement` -> `inelastic_strain_increment`.